### PR TITLE
updated README

### PR DESCRIPTION
**Pull request by gurgeous**
How about this for #39? Specifically:

1. Reordered sections, some wordsmithing
1. Redid FEATURES section
1. Badge
1. Code fences
1. Moved API to the bottom and renamed to REFERENCE

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,140 +1,111 @@
...
+
 # command_kit
 
+A Ruby toolkit for building clean, correct, and robust CLI commands as
+plain-old Ruby classes.
+
 * [Homepage](https://github.com/postmodern/command_kit.rb#readme)
-* [Forum](https://github.com/postmodern/command_kit.rb/discussions)
-* [Issues](https://github.com/postmodern/command_kit.rb/issues)
+* [Forum](https://github.com/postmodern/command_kit.rb/discussions) / [Issues](https://github.com/postmodern/command_kit.rb/issues)
```
Comment (postmodern) [inline README.md:9]: This is a good idea. :+1: Maybe use a `|` character.

File: README.md [hunk 2]
```diff
@@ -1,140 +1,111 @@
+[![Build Status](https://github.com/postmodern/command_kit.rb/workflows/CI/badge.svg?branch=main)](https://github.com/postmodern/command_kit.rb/actions)
```
Comment (postmodern) [inline README.md:1]: Normally I put the badges below the README `# Project` heading.

File: README.md [hunk 3]
```diff
@@ -1,140 +1,111 @@
...
+plain-old Ruby classes.
+
 * [Homepage](https://github.com/postmodern/command_kit.rb#readme)
-* [Forum](https://github.com/postmodern/command_kit.rb/discussions)
-* [Issues](https://github.com/postmodern/command_kit.rb/issues)
+* [Forum](https://github.com/postmodern/command_kit.rb/discussions) / [Issues](https://github.com/postmodern/command_kit.rb/issues)
 * [Documentation](http://rubydoc.info/gems/command_kit/frames)
 
-## Description
+## Install
```
Comment (postmodern) [inline README.md:12]: Eh, I tend to try to sell the reader on the project first then show how to install it. (Description -> Features -> Example Code -> Install). Maybe Examples could go below Install.

File: README.md [hunk 4]
```diff
@@ -1,140 +1,111 @@
...
-* [CommandKit::Usage](https://rubydoc.info/gems/command_kit/CommandKit/Usage)
-* [CommandKit::XDG](https://rubydoc.info/gems/command_kit/CommandKit/XDG)
+* **Simplicity** - Commands are plain-old ruby classes, with options and arguments as attributes. Extend commands via inheritance. Uses [OptionParser][optparse] for POSIX option parsing.
+* **Subcommands** - Supports subcommands (explicit or lazy-loaded) and command aliases.
+* **Signals and errors** - Handles Ctrl^C and SIGINT interrupts (aka exit 130). Handles broken pipes (aka `mycmd | head`).
+* **Redirects** - Correctly handles when stdout or stdin is redirected to a file.
+* **Man** - Display a man-page instead of `--help` (see {CommandKit::Help::Man}).
+* **UX** - Optional ANSI coloring. Interactive prompt for user input. Supports easily detecting the terminal size, and paging output with `less` or `more`.
+* **XDG** - Supports [XDG directories](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) (`~/.config/`, `~/.local/share/`, `~/.cache/`).
+* **Testability** - Easy to test (ex: `MyCmd.main(arg1, arg2, options: {foo: foo}) # => 0`)
```
Comment (postmodern) [inline README.md:31]: The bolded keywords are a really good idea. :+1:

File: README.md [hunk 5]
```diff
@@ -1,140 +1,111 @@
...
-* [CommandKit::Stdio](https://rubydoc.info/gems/command_kit/CommandKit/Stdio)
-* [CommandKit::Terminal](https://rubydoc.info/gems/command_kit/CommandKit/Terminal)
-* [CommandKit::Usage](https://rubydoc.info/gems/command_kit/CommandKit/Usage)
-* [CommandKit::XDG](https://rubydoc.info/gems/command_kit/CommandKit/XDG)
+* **Simplicity** - Commands are plain-old ruby classes, with options and arguments as attributes. Extend commands via inheritance. Uses [OptionParser][optparse] for POSIX option parsing.
+* **Subcommands** - Supports subcommands (explicit or lazy-loaded) and command aliases.
+* **Signals and errors** - Handles Ctrl^C and SIGINT interrupts (aka exit 130). Handles broken pipes (aka `mycmd | head`).
+* **Redirects** - Correctly handles when stdout or stdin is redirected to a file.
+* **Man** - Display a man-page instead of `--help` (see {CommandKit::Help::Man}).
+* **UX** - Optional ANSI coloring. Interactive prompt for user input. Supports easily detecting the terminal size, and paging output with `less` or `more`.
```
Comment (postmodern) [inline README.md:29]: Although it seems a little odd how we have concepts like Simplicity or UX along side things like Man and XDG. Maybe we should have a sub-list called "Standards" or "Integration" and list ANSI, XDG, etc.

**Review by postmodern — CHANGES_REQUESTED** [review]
Some things I like, other things I'm not sold on. Maybe split up into multiple commits so I can cherry-pick in individual changes, like the code fences.

Also, weren't we going to copy in all of the Example code from each module into the Examples section?

**Comment by gurgeous** [comment]
> Some things I like, other things I'm not sold on. Maybe split up into multiple commits so I can cherry-pick in individual changes, like the code fences.
> 
> Also, weren't we going to copy in all of the Example code from each module into the Examples section?

Let me take one more stab at it. It's all in one file and divided into sections so it's easy to make more changes. Not attached to any particular change. Hardest part is convincing VS Code not to autoformat 😛 

I'll do this:

1. Let's add more examples in a separate PR. I need to read those examples first.
1. Reorder badge/install/etc per feedback
1. Try again with bolded features (Standards, Integration, etc.)

**Comment by gurgeous** [comment]
How about this? I have no ego on this, feel free to keep commenting and I'll keep messing with it

**Comment by postmodern** [comment]
Doing some additional edits to your PR before squash-merge-signoff. Trying to think of a good word to describe all of the additional features, which can be optionally included in if you need them. Featureful? Fully Featured? Fully Loaded? Kitchen Sink? Swiss Army Knife?

**Comment by gurgeous** [comment]
I like Kitchen Sink! "This stuff is useful and you might want it" but also tongue in cheek. That's usually how I try to communicate anyway 😄

**Comment by postmodern** [comment]
OK manually squash merged the commits with my edits in 9831595106320260f0eebe2506a4fd64e891aeba. Was hoping that git would preserve both our authorships, but also don't want to spend too much time on the README. I rewrote the Features sections to have better Key Points (Simple, Correct, Complete, Testable) and reorganized some of the sections (Requirements should be next to Install, imo).

**Pull request closed without merging**